### PR TITLE
add: lazy loading modal solana selector

### PR DESCRIPTION
**Pull request by LukassF**

### CODE REVIEW

**Review by NorbertBodziony — CHANGES_REQUESTED** [review]

File: sdk/bindings/WalletMetadata.ts [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+  chains?: Array<Network>
+  version?: Version
+  walletType: WalletType
+  mobile: Deeplink | null
+  desktop: Deeplink | null
+  image: Images
+  app?: Record<Platform, string>
+  injectPath?: Record<Network, string>
+  lastUpdatedTimestamp?: bigint
+}
```
Comment (NorbertBodziony) [inline sdk/bindings/WalletMetadata.ts:23]: DO NOT FORMAT GENERATE FILES

File: sdk/packages/selector-polkadot/src/detection.ts [hunk 2]
```diff
@@ -42,7 +42,10 @@ export const getPolkadotWalletsList = (presetList: WalletMetadata[], recentWalle
   const walletsData: Record<string, IPolkadotWalletListItem> = {}
 
   presetList.forEach((wallet) => {
-    walletsData[wallet.slug.toLocaleLowerCase()] = wallet
+    walletsData[wallet.slug.toLocaleLowerCase()] = {
+      ...wallet,
+      recent: recentWalletName === wallet.name
```
Comment (NorbertBodziony) [inline sdk/packages/selector-polkadot/src/detection.ts:47]: is recent part of api ?

**Review by NorbertBodziony — CHANGES_REQUESTED** [review]

File: sdk/packages/selector-base/src/types.ts [hunk 3]
```diff
@@ -19,6 +19,15 @@ export interface MetadataWallet {
...
 
+export interface WalletMetadata extends Partial<WalletMetadataOriginal> {
+  slug: string
+  name: string
+  walletType: WalletType
+  mobile: Deeplink | null
+  desktop: Deeplink | null
+  image: Images
+}
+
```
Comment (NorbertBodziony) [inline sdk/packages/selector-base/src/types.ts:30]: WHAT ? remove this

File: sdk/packages/selector-base/src/types.ts [hunk 4]
```diff
@@ -2,8 +2,8 @@ import { type AppBaseInitialize } from '@nightlylabs/nightly-connect-base'
 import { type Deeplink } from '@nightlylabs/nightly-connect-base/dist/types/bindings/Deeplink'
 import { type Wallet } from '@wallet-standard/core'
 import { type WalletType } from '../../../bindings/WalletType'
-import { type WalletMetadata } from '../../../bindings/WalletMetadata'
-export { type WalletMetadata } from '../../../bindings/WalletMetadata'
+import { type WalletMetadata as WalletMetadataOriginal } from '../../../bindings/WalletMetadata'
```
Comment (NorbertBodziony) [inline sdk/packages/selector-base/src/types.ts:5]: just export this

File: sdk/packages/selector-base/src/detection.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { Wallet, getWallets } from '@wallet-standard/core'
-import { WalletMetadata, type IWalletListItem, type MetadataWallet } from './types'
+import { type WalletMetadata, type IWalletListItem, type MetadataWallet } from './types'
 
 export const getWalletsList = (
   presetList: WalletMetadata[],
```
Comment (NorbertBodziony) [inline sdk/packages/selector-base/src/detection.ts:5]: we gonna remove this method each selecto will define it for itself

File: sdk/packages/selector-polkadot/src/detection.ts [hunk 6]
```diff
@@ -32,7 +31,9 @@ export const getPolkadotWallets = (): PolkadotWalletInjected[] => {
   }
 }
 
-export interface IPolkadotWalletListItem extends Omit<IWalletListItem, 'standardWallet'> {
+export interface IPolkadotWalletListItem extends WalletMetadata {
+  recent?: boolean
+  detected?: boolean
   injectedWallet?: PolkadotWalletInjected
 }
```
Comment (NorbertBodziony) [inline sdk/packages/selector-polkadot/src/detection.ts:38]: WalletMetadata should be one from bindings 
so you almost got it

**Review by NorbertBodziony — CHANGES_REQUESTED** [review]

File: sdk/packages/selector-base/src/types.ts (hunk 3 above)
Comment (LukassF) [reply sdk/packages/selector-base/src/types.ts:30]: But wouldnt it then force me to pass all of the fields such ass homepage, version, app etc. in for example polkadot selector detection file?
`walletsData[wallet.name.toLocaleLowerCase()] = {
      slug: wallet.name,
      name: wallet.name,
      image: {
        default: wallet.icon as string,
        sm: wallet.icon as string,
        md: wallet.icon as string,
        lg: wallet.icon as string
      },
      desktop: null,
      mobile: null,
      recent: recentWalletName === wallet.name,
      detected: true,
      injectedWallet: wallet,
      walletType: 'hybrid'
    }`

Comment (LukassF) [reply sdk/packages/selector-base/src/types.ts:30]: But wouldnt it then force me to pass all of the fields such as homepage, version, app etc. in for example polkadot selector detection file?
`    walletsData[wallet.name.toLocaleLowerCase()] = {
      slug: wallet.name,
      name: wallet.name,
      image: {
        default: wallet.icon as string,
        sm: wallet.icon as string,
        md: wallet.icon as string,
        lg: wallet.icon as string
      },
      desktop: null,
      mobile: null,
      recent: recentWalletName === wallet.name,
      detected: true,
      injectedWallet: wallet,
      walletType: 'hybrid'
    }`

File: sdk/packages/selector-polkadot/src/adapter.ts [hunk 7]
```diff
@@ -163,7 +161,7 @@ export class NightlyConnectAdapter implements Injected {
     )
     if (!adapter._connectionOptions.disableModal) {
       adapter._modal = new NightlyConnectSelectorModal(
-        adapter.walletsList,
+        adapter.walletsList as IWalletListItem[],
```
Comment (NorbertBodziony) [inline sdk/packages/selector-polkadot/src/adapter.ts:164]: do we need as also 
why us `IWalletListItem`

File: sdk/packages/selector-solana/src/adapter.ts [hunk 8]
```diff
@@ -137,445 +140,488 @@ export class NightlyConnectAdapter extends BaseMessageSignerWalletAdapter {
...
+                  } catch {
+                    this.disconnect()
+                  }
+                })
+              }
+              // timeout after 5 seconds
+              if (checks > 500) {
+                clearInterval(loadingInterval)
+                // reject(new Error('Connecting takes too long'))
+                // TODO we need to have a way to show error on modal
```
Comment (NorbertBodziony) [inline sdk/packages/selector-solana/src/adapter.ts:591]: show error ?

**Review by NorbertBodziony — CHANGES_REQUESTED** [review]

File: sdk/packages/selector-polkadot/src/adapter.ts [hunk 9]
```diff
@@ -526,42 +526,42 @@ export class NightlyConnectAdapter
...
-                  this.emit('connect', await this.accounts.get())
-                } catch {
-                  this.disconnect()
-                }
+                })
+                this._loading = false
+              })
+              .catch(() => {
+                this._loading = false
+                throw new Error('Failed to initialize adapter')
```
Comment (NorbertBodziony) [inline sdk/packages/selector-polkadot/src/adapter.ts:563]: we should reject not throw

**Review by NorbertBodziony — CHANGES_REQUESTED** [review]

**Pull request merged**